### PR TITLE
Js_of_ocaml frontend. Site restructuring 

### PR DESCRIPTION
**Pull request by hernoufM**
This is a big PR that adds principally 2 changes: 

1. New type of front-end written on **js_of_ocaml** that implements all required features.
 It can be used by pages by specifying option ```--frontend js_ocaml```. Frontend library depends on API server (named **search-api**) that process requests from js_of_ocaml frontend. It should be pinned with 
```bash
opam pin https://github.com/hernoufM/digodoc-search-api.git
```
Requests are implemented with library **ez_api.ixhr_lwt**. This library makes uses of some packages that are defined as optional dependencies to **ez_api**  which means
that they are not installed automatically by opam. Install it with : 
```bash
opam install  geoip js_of_ocaml-lwt calendar tls lwt_log  cohttp-lwt-jsoo ppxlib  
``` 
2. Site restructuring, that brings together all scripts, style's files, docs, index pages, etc.

### CODE REVIEW

File: res.txt [hunk 1]
```diff
@@ -0,0 +1,288 @@
+sending incremental file list
```
Comment (zapashcanon) [inline res.txt:1]: I believe this file has been added by mistake. :)

File: src/digodoc_common/main.ml [hunk 2]
```diff
@@ -10,7 +10,8 @@
...
 
-(* options that are modified only using digodoc *)
-let sources = ref true
 
-let with_header = ref false
\ No newline at end of file
+(* If you delete or rename this file, you should add
+   'src/digodoc_common/main.ml' to the 'skip' field in "drom.toml" *)
+
+let main () = Printf.printf "Hello world!\n"
```
Comment (zapashcanon) [inline src/digodoc_common/main.ml:17]: I believe this is a mistake.

File: res.txt (hunk 1 above)
Comment (hernoufM) [reply res.txt:1]: Oops :)

File: src/digodoc_common/main.ml (hunk 2 above)
Comment (hernoufM) [reply src/digodoc_common/main.ml:17]: This library was generated automatically with library skeleton by _drom_. But I confirm, this file doesn't have any sens

**Pull request merged**